lua/tardis/sh_interiors.lua

```diff
@@ -76,47 +96,130 @@ function TARDIS:SaveIntPreferences(preferences)
+-- Interior custom settings
+
+function TARDIS:GetMainVersionId(int_id)
+    return (self.Metadata[int_id] and self.Metadata[int_id].IsVersionOf) or int_id
 end
 
-function TARDIS:IsUnwantedInt(id, ply)
-    return TARDIS:GetIntPreferences(ply).unwanted[id]
+function TARDIS:GetIntCustomSetting(int_id, setting_id, ply, default_val)
+    local int_id = TARDIS:GetMainVersionId(int_id)
+    local metadata = self.Metadata[int_id]
+
+    local int_pref = TARDIS:GetIntPreferences(ply)
+    if not int_pref or not int_pref.custom_settings then return default_val end
+
+    local settings = int_pref.custom_settings[int_id]
+    if settings and settings[setting_id] ~= nil then
+        return settings[setting_id]
+    end
+
+    if setting_id == "preferred_spawn" then
+        return TARDIS:SetupPreferredSpawn(int_id, ply)
+    end
+
+    local md_settings = metadata.CustomSettings
+    if md_settings and md_settings[setting_id] and md_settings[setting_id].value ~= nil then
+        local metadata_value = md_settings[setting_id].value
+        TARDIS:SetIntCustomSetting(int_id, setting_id, metadata_value, ply)
+        return metadata_value
+    end
+
+    return default_val
 end
 
-function TARDIS:SetFavoriteInt(id, favorite, ply)
-    if favorite == false then favorite = nil end -- clean up (mainly for debugging purposes)
+function TARDIS:SetIntCustomSetting(int_id, setting_id, value, ply)
+    local int_id = TARDIS:GetMainVersionId(int_id)
 
-    local interior_preferences = TARDIS:GetIntPreferences(ply)
-    interior_preferences.favorites[id] = favorite
-    self:SaveIntPreferences(interior_preferences)
+    local int_pref = TARDIS:GetIntPreferences(ply)
+    int_pref.custom_settings = int_pref.custom_settings or {}
+    int_pref.custom_settings[int_id] = int_pref.custom_settings[int_id] or {}
+    int_pref.custom_settings[int_id][setting_id] = value
+    self:SaveIntPreferences(int_pref)
 end
 
-function TARDIS:SetUnwantedInt(id, unwanted, ply)
-    if unwanted == false then unwanted = nil end -- clean up (mainly for debugging purposes)
+function TARDIS:ResetIntCustomSettings(ply, int_id)
+    local int_id = TARDIS:GetMainVersionId(int_id)
+
+    local int_pref = TARDIS:GetIntPreferences(ply)
+    if not int_pref then return end
 
-    local interior_preferences = TARDIS:GetIntPreferences(ply)
-    interior_preferences.unwanted[id] = unwanted
-    self:SaveIntPreferences(interior_preferences)
+    if int_id then
+        int_pref.custom_settings = int_pref.custom_settings or {}
+        int_pref.custom_settings[int_id] = {}
+    else
+        int_pref.custom_settings = {}
+    end
+
+    self:SaveIntPreferences(int_pref)
+end
+
+function TARDIS:ToggleIntCustomSetting(int_id, setting_id, ply)
+    local int_id = TARDIS:GetMainVersionId(int_id)
+
+    local value = TARDIS:GetIntCustomSetting(int_id, setting_id, ply)
+    TARDIS:SetIntCustomSetting(int_id, setting_id, (not value), ply)
+end
+
+function TARDIS:SetupPreferredSpawn(int_id, ply)
```

Comment:
Maybe `TARDIS:SetupPreferredVersion` would make more sense?

Reply:
I was afraid to cause confusion there, since a `version` is already a thing 
(as "versions" which can be chosen on the right in the spawnmenu, either a named id or a combination of two IDs)

Reply:
Nevermind, renamed it